File: lista-de-exercicios/exercicio-3/Jogador.test.ts
import { jest } from '@jest/globals';
import { Escolhas } from './Jokenpo.js';
import { Jogador } from './Jogador';

const pedra = Escolhas.PEDRA;
const papel = Escolhas.PAPEL;
const tesoura = Escolhas.TESOURA;

describe('Classe Jogador', () => {
    test('A escolha do jogador deve ser pedra, papel ou tesoura', () => {
        const jogadorPedra = new Jogador(pedra);

        expect(jogadorPedra.getEscolha()).toBe(pedra);

        const jogadorPapel = new Jogador(papel);

        expect(jogadorPapel.getEscolha()).toBe(papel);

        const jogadorTesoura = new Jogador(tesoura);

        expect(jogadorTesoura.getEscolha()).toBe(tesoura);
    })
})

File: lista-de-exercicios-java/src/test/java/org/example/exercicio03/JogadorTest.java
package org.example.exercicio03;

import static org.example.exercicio03.Escolha.PEDRA;
import static org.example.exercicio03.Escolha.PAPEL;
import static org.example.exercicio03.Escolha.TESOURA;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class JogadorTest {
    @DisplayName("A escolha do jogador deve ser pedra, papel ou tesoura")
    @Test
    void escolhaJogador() {
        Jogador jogadorPedra = new Jogador(PEDRA);
        Jogador jogadorPapel = new Jogador(PAPEL);
        Jogador jogadorTesoura = new Jogador(TESOURA);

        assertEquals(PEDRA, jogadorPedra.getEscolha());
        assertEquals(PAPEL, jogadorPapel.getEscolha());
        assertEquals(TESOURA, jogadorTesoura.getEscolha());
    }
}